pkg/gateway/middleware/deployment_route.go

```diff
@@ -23,7 +23,7 @@ func (f FindDeploymentRouteMiddleware) Handle(next http.Handler) http.Handler {
 
 		match := model.MatchRoute(path, app.Config.DeploymentRoutes)
 		if match == nil {
-			http.Error(w, "Fail to match deployment route", http.StatusNotFound)
+			http.Error(w, "Not found", http.StatusNotFound)
```

Comment:
I think we should keep the detail like `Deployment route not found`? If developer see the message `Not found`, there will be chance that he call a wrong app ("App not found") or wrong path ("Deployment route not found"). More detail may be helpful for debugging?

Reply:
I'm not sure if using richer message would cause potential information disclosure problem. For example in nginx, generic 404 error would be returned regardless of whether the host/path is not matched.

In my opinion:
- Errors like `Deployment route not found` are not user-friendly, since these terms is unfamiliar to developers, and may mislead developer.
- These errors can be simply resolved by double-checking the URL, so developer should be able to troubleshoot it easily. For advanced cases (e.g. path matching problems), additional information would not help.